server/src/main/java/org/elasticsearch/action/delete/DeleteRequest.java

```diff
@@ -83,13 +84,13 @@ public DeleteRequest(String index, String type, String id) {
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = super.validate();
-        if (type == null) {
+        if (Strings.hasLength(type) == false) {
             validationException = addValidationError("type is missing", validationException);
         }
-        if (id == null) {
+        if (Strings.hasLength(id) == false) {
```

Comment:
nit: maybe use Strings.isEmpty or hasText?

Reply:
sure, thanks @dnhatn I addressed that in 1d54c0e